examples/simple_example/main.tf

```diff
@@ -54,13 +79,15 @@ module "dataflow-job" {
 module "dataflow-job-2" {
   source                = "../../"
   project_id            = "${var.project_id}"
-  name              = "wordcount-terraform-example-2"
+  name                  = "wordcount-terraform-example-2"
   on_delete             = "cancel"
   zone                  = "${var.region}-a"
   max_workers           = 1
   template_gcs_path     = "gs://dataflow-templates/latest/Word_Count"
   temp_gcs_location     = "${module.dataflow-bucket.name}"
   service_account_email = "${var.service_account_email}"
+  network               = "${module.vpc.network_name}"
+  subnetwork            = "regions/${module.vpc.subnets_regions[0]}/subnetworks/${module.vpc.subnets_names[0]}"
```

Comment:
Given that the network module is only capable of outputting self_links, would it be possible to accept self_links instead of this format? On every module I've written for customers that uses a combination of our network module with anything else, I've had to add a replace statement to the network_self_link output.

Reply:
@Tfmenard Let's do this.

Reply:
@thefirstofthe300 
Just added it for the subnetwork argument here.
7d5b02b4d627c388fed3146f36062827927231fa
and here 
5b9f2cbc01e77e023675bff2142c077a9a69d71f

Reply:
TBH, I prefer accepting the self link and parsing out what is needed. Less chance of needing to change the module API to either accept a new network related value later on down the road. Plus, it's more symmetrical since the subnetwork argument accepts a self link.

Reply:
Agreed. I had already implemented replace statements to accept self links for both network and subnetwork so I reverted back to these changes here ffa3bfde26b32a78d8a98ccf6f1880e6e059e90a